src/components/signup/SignupState.js

```diff
@@ -521,13 +500,13 @@ const Signup = ({ navigation }: { navigation: any, screenProps: any }) => {
     if (nextRoute === undefined) {
       const ok = await waitForRegistrationToFinish()
 
-      //this will cause a re-render and move user to the dashboard route
+      // this will cause a re-render and move user to the dashboard route
       if (ok) {
-        store.set('isLoggedIn')(true)
+        setSuccessfull(() => store.set('isLoggedIn')(true))
       }
     } else if (nextRoute && nextRoute.key === 'SMS') {
       try {
-        const result = await checkExisting({ mobile: newState.mobile })
+        const result = await checkExisting(torusProvider, { mobile: newState.mobile }, { fromSignupFlow: true })
 
         if (result !== 'signup') {
```

Comment:
if you chose 'continue' with signup - the flow should continue here (there will be no return), never mind are you on self custody or not